### PR TITLE
KEYCLOAK-11747 Add API credentials tests

### PR DESCRIPTION
**Pull request by fperot74**

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java [hunk 1]
```diff
@@ -1569,19 +1575,91 @@ public void loginShouldFailAfterPasswordDeleted() {
...
         driver.navigate().to(accountUrl);
         assertEquals("Test user should be on the login page.", "Log In", PageUtils.getPageTitle(driver));
         loginPage.login(userName, userPass);
-        assertTrue("Test user should fail to log in after password was deleted.", 
+        assertTrue("Test user should fail to log in after password was deleted.",
                 driver.getCurrentUrl().contains(String.format("/realms/%s/login-actions/authenticate", REALM_NAME)));
     }
-    
+
+    private UserResource findUser(String realm, String username) {
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java]: I think it is possible to remove this method and use already existing method "ApiUtil.findUserByUsername" ?

**Review by mposolda — COMMENTED** [review]
@fperot74 Thanks, I've added one minor comment about util method, which we already have.

Also I think that there will be conflict with the PR https://github.com/cloudtrust/keycloak/pull/31 , so either the first or second PR will need to be slightly adjusted depending on which would go first.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java (hunk 1 above)
Comment (fperot74) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java]: Removed

**Review by mposolda — APPROVED** [review]
@fperot74 Thanks, approving. My comment with conflict is still valid, but will need to be fixed later either in this PR or in the second PR by @tkyjovsk

**Review by AlistairDoswald — APPROVED** [review]
Approving. I'll correct the method name changes in a second commit.

**Pull request merged**